src/client/api/tests/job_tests.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * (C) Copyright 2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. 8F-30005.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+/**
+ * Unit tests for the job API for the client lib
+ */
+
+#include <stdarg.h>
+#include <stdlib.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <daos_types.h>
+#include <daos_errno.h>
+#include <daos/common.h>
+#include <daos/job.h>
+#include <string.h>
+#include <sys/utsname.h>
+
+
+/*
+ * Mocks
+ */
+
+static char *getenv_daos_jobid_return; /* value returned for DAOS_JOBID */
+static char *getenv_jobid_env_return; /* value returned for DAOS_JOBID_ENV */
+static char *getenv_jobid_return; /* value returned for value of DAOS_JOBID_ENV */
+char *getenv(const char *name)
+{
+	if (strncmp(name, DEFAULT_JOBID_ENV, sizeof(DEFAULT_JOBID_ENV)) == 0) {
+		return getenv_daos_jobid_return;
+	} else if (strncmp(name, JOBID_ENV, sizeof(JOBID_ENV)) == 0 ) {
+		return getenv_jobid_env_return;
+	} else if (getenv_jobid_env_return &&
+		strncmp(name, getenv_jobid_env_return, MAX_ENV_NAME) == 0) {
+		return getenv_jobid_return;
+	} else {
+		return NULL;
+	}
+}
+
+static pid_t getpid_pid;
+pid_t getpid(void)
+{
+	return getpid_pid;
+}
+
+static char *uname_nodename;
+int uname (struct utsname *buf)
+{
+	if (!buf) {
+		errno = EFAULT;
+		return -1;
+	}
```

Comment:
You could also add a test to exercise the uname error handling. You'd enable that by having a uname_errno and uname_return, which would let you arbitrarily return an error from this mock uname, without having to pass in anything special.